src/util/map.js

```diff
@@ -122,3 +122,26 @@ export function mapEntries<K, V, InK, InV>(
   }
   return result;
 }
+
+/**
+ * Concatenate maps without mutating the arguments.
+ *
+ * Combines multiple maps, returning a new map which has every key from
+ * the source maps, with their corresponding values. None of the inputs
+ * are mutated. In the event that multiple maps have the same key, an
+ * error will be thrown.
+ */
+export function concat<K, V>(maps: $ReadOnlyArray<Map<K, V>>): Map<K, V> {
+  const result = new Map();
+  let updates = 0;
+  for (const map of maps) {
+    for (const [key, value] of map.entries()) {
+      updates++;
```

Comment:
Note that you could find exactly which key was the duplicate without a
`has`-check at each iteration, by making sure that `result.size`
increments by exactly one at each iteration instead of checking it at
the end. This requires *n* accesses to `size`, which is certainly
cheaper than *n* calls to `has`, and probably free in practice.

I have no strong preference, so I’ll leave it up to you.

Reply:
Done, thanks.